src/components/GlobalDataProvider.tsx

```diff
@@ -23,11 +23,29 @@ const GlobalDataProvider = ({ children }: { children: ReactElement }) => {
       const { Ticker } = await arioReadSDK.getInfo();
       setTicker(Ticker);
 
-      const currentEpoch = await arioReadSDK.getCurrentEpoch();
-      setCurrentEpoch(currentEpoch);
+      try {
+        const currentEpoch = await arioReadSDK.getCurrentEpoch();
+
+        // FIXME: This is here to prevent the app from crashing when the current epoch comes back as an empty array.
+        // This is due to how contract and SDK are currently handling the epoch data situation when it can't be fetched.
+        // This should be removed when the above situation is changed to throw an exception.
+        if (Array.isArray(currentEpoch)) {
```

Comment:
you could do a falsey check here - to protect against undefined or empty objects instead of just arrays

Reply:
Will update that as part of a PR I'm working on for v1.5.0

Reply:
Following up here, I checked and empty arrays and objects are truthy in JS, so we'd need to check against various conditions explicitly.